geonode/base/auth.py

```diff
@@ -21,16 +21,35 @@
 import datetime
 import logging
 import traceback
-
 from django.utils import timezone
 from django.conf import settings
 from django.contrib.auth import authenticate
 from oauth2_provider.models import AccessToken, get_application_model
 from oauthlib.common import generate_token
+from guardian.shortcuts import get_anonymous_user
 
 logger = logging.getLogger(__name__)
 
 
+def extract_user_from_headers(request):
+    if not request.user.is_authenticated or request.user == get_anonymous_user():
```

Comment:
I would propose to change `request.user == get_anonymous_user()` with `request.user.is_anonymous`